### PR TITLE
Add missing register class conversion for inline asm

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Part of https://github.com/rust-lang/rustc_codegen_gcc/issues/60.

### CODE REVIEW

File: src/asm.rs [hunk 1]
```diff
@@ -564,56 +576,77 @@ fn reg_to_gcc(reg: InlineAsmRegOrRegClass) -> ConstraintOrRegister {
...
             }
         },
+        // They can be retrieved from https://gcc.gnu.org/onlinedocs/gcc/Machine-Constraints.html
         InlineAsmRegOrRegClass::RegClass(reg) => match reg {
-            InlineAsmRegClass::AArch64(AArch64InlineAsmRegClass::preg) => unimplemented!(),
-            InlineAsmRegClass::AArch64(AArch64InlineAsmRegClass::reg) => unimplemented!(),
-            InlineAsmRegClass::AArch64(AArch64InlineAsmRegClass::vreg) => unimplemented!(),
-            InlineAsmRegClass::AArch64(AArch64InlineAsmRegClass::vreg_low16) => unimplemented!(),
-            InlineAsmRegClass::Arm(ArmInlineAsmRegClass::reg) => unimplemented!(),
+            InlineAsmRegClass::AArch64(AArch64InlineAsmRegClass::reg) => {
```
Comment (Amanieu) [inline src/asm.rs:581]: This should just be `"r"`. `L` and `K` are for integer constants, not registers.

Comment (GuillaumeGomez) [reply src/asm.rs]: Thanks, fixed!

File: src/base.rs [hunk 2]
```diff
@@ -112,7 +112,9 @@ pub fn compile_codegen_unit<'tcx>(tcx: TyCtxt<'tcx>, cgu_name: Symbol, supports_
         context.add_command_line_option("-mavx");
 
         for arg in &tcx.sess.opts.cg.llvm_args {
-            context.add_command_line_option(arg);
+            if !arg.starts_with("--x86-asm-syntax=") {
+                context.add_command_line_option(arg);
+            }
```
Comment (bjorn3) [inline src/base.rs:117]: This would break anyone that needs a specific assembly syntax in rustc's output, right?

Comment (antoyo) [reply src/base.rs:117]: We're currently doing some tests.

This option is an LLVM option; for gcc it should be `-masm=intel`. I'm not sure what's the plan for supporting setting options for the different backends, but cg_gcc currently uses `llvm-args` to set gcc options. Since this option is to send arguments to the backend, I'm not sure it makes sense to do the mapping between the LLVM options and GCC options.

I guess we would need to discuss this somewhere to agree on the correct solution, though.

File: src/asm.rs [hunk 3]
```diff
@@ -691,7 +706,10 @@ impl<'gcc, 'tcx> AsmMethods<'tcx> for CodegenCx<'gcc, 'tcx> {
             && options.contains(InlineAsmOptions::ATT_SYNTAX);
 
         // Build the template string
-        let mut template_str = String::new();
+        let mut template_str = ".pushsection .text\n".to_owned();
```
Comment (antoyo) [inline src/asm.rs:709]: Note to self: check why the change to these directives was done in the past to make sure this change is correct.

Comment (GuillaumeGomez) [reply src/asm.rs:709]: This change itself is just to make the code easier to follow, it still outputs the same thing.

**Review by antoyo — APPROVED** [review]
Thanks a lot!
